src/bpftrace.cpp

```diff
@@ -1673,6 +1686,52 @@ std::string BPFtrace::resolve_uid(uintptr_t addr) const
   return username;
 }
 
+std::string BPFtrace::resolve_timestamp(uint32_t strftime_id,
+                                        uint64_t nsecs_since_boot)
+{
+  auto fmt = strftime_args_[strftime_id].c_str();
+  char timestr[STRING_SIZE];
+  struct tm tmp;
+  std::ifstream file("/proc/stat");
+  if (!file)
+  {
+    std::cerr << "Fail to open file /proc/stat: " << std::strerror(errno)
+              << std::endl;
+    return "(?)";
+  }
+  std::string line, field;
+  uint64_t btime = 0;
+  while (std::getline(file, line))
+  {
+    std::stringstream ss(line);
+    ss >> field;
+    if (field == "btime")
+    {
+      ss >> btime;
+      if (ss.fail())
+        btime = 0;
+      break;
+    }
+  }
+  if (btime == 0)
```

Comment:
btime should be a "kernel constant", making it possible to read its value once and storing it (maybe before code gets loaded into the kernel even). That would remove some useless string parsing from the "hot path" .

Reply:
Sure, will do.